src/librustc/mir/interpret/error.rs

```diff
@@ -344,12 +355,23 @@ pub enum InterpError<'tcx> {
     ReadUndefBytes(Size),
     DeadLocal,
     InvalidBoolOp(mir::BinOp),
+    InlineAsm,
+    UnimplementedTraitSelection,
+    CalledClosureAsFunction,
+    NoMirFor(String),
+    /// This variant is used by machines to signal their own errors that do not
+    /// match an existing variant.
+    MachineError(String),
+
+    /// Not actually an interpreter error -- used to signal that execution has exited
+    /// with the given status code.  Used by Miri, but not by CTFE.
+    Exit(i32),
+
     Unimplemented(String),
```

Comment:
Could you move this variant to the top of this enum (it is most representative of the `UnsupportedInfo`)?

Reply:
Done. Now do we wait until we are clear about how to move stuff from `UnsupportedInfo`?

Reply:
I would say for this PR, we leave it all in there.